### PR TITLE
fix: Upgrading mapbox version

### PR DESCRIPTION
**Pull request by jorelosorio**
It fixes an issue regarding rotation after discussion with @marius-at-atb 

Closes https://github.com/AtB-AS/kundevendt/issues/3688

### CODE REVIEW

**Comment by jorelosorio** [comment]
This change introduces another issues regarding `Mapbox"' has no exported member 'Padding'` and other, it will be on hold until have more time to migrate maps.

File: src/tariff-zones-selector/TariffZonesSelectorMap.tsx [hunk 1]
```diff
@@ -182,7 +183,7 @@ const TariffZonesSelectorMap = ({
             </MapboxGL.ShapeSource>
             {featureCollection.features.map((f) => (
               <MapboxGL.ShapeSource
-                key={f.id}
+                key={typeof f.id === 'number' ? f.id?.toString() : f.id}
```
Comment (KristianSelnas) [inline src/tariff-zones-selector/TariffZonesSelectorMap.tsx:186]: This could be simplified to `String(id)`. Not important at all, just a tip for later

File: src/components/map/types.ts [hunk 2]
```diff
@@ -38,6 +38,8 @@ export type MapRegion = {
   center: Position;
 };
 
+export type MappPadding = number | number[];
```
Comment (KristianSelnas) [inline src/components/map/types.ts:41]: `Mapp` -> `Map`

Comment (KristianSelnas) [inline src/components/map/types.ts:41]: Why do you need this type? It seems a bit "dangerous" as the original type (MapboxGL.Padding) enforces the correct number of numbers in the array. This doesn't, and that could cause run-time errors that otherwise would have been discovered at compile time.

File: src/components/map/MapConfig.ts [hunk 3]
```diff
@@ -1,8 +1,9 @@
-import {CameraStop, MapViewProps} from '@rnmapbox/maps';
+//import {CameraStop, MapViewProps} from '@rnmapbox/maps';
 import {Platform} from 'react-native';
 import {MAPBOX_STOP_PLACES_STYLE_URL} from '@env';
+import {CameraStop} from '@rnmapbox/maps/lib/typescript/components/Camera';
 
-export const MapViewConfig: MapViewProps = {
```
Comment (KristianSelnas) [inline src/components/map/MapConfig.ts:5]: Why did you remove this?

File: src/components/map/MapConfig.ts [hunk 4]
```diff
@@ -1,8 +1,9 @@
-import {CameraStop, MapViewProps} from '@rnmapbox/maps';
+//import {CameraStop, MapViewProps} from '@rnmapbox/maps';
```
Comment (KristianSelnas) [inline src/components/map/MapConfig.ts:1]: If you have good reasons for removing the type for `MapViewConfig` you should remove his commented out import as well

File: src/components/map/types.ts (hunk 2 above)
Comment (jorelosorio) [reply src/components/map/types.ts:41]: Hi, that is how the parameter now is expected from the library before it was Padding but is removed:

image

I just created a MapPadding to do not repeat on all the places we use it. 

I will fix the typo ;) thanks

File: src/components/map/MapConfig.ts (hunk 4 above)
Comment (jorelosorio) [reply src/components/map/MapConfig.ts:1]: Thanks I will fix it good catch ;), I haven't properly double-checked it just a small draft to let @marius-at-atb test his changes.

File: src/components/map/types.ts (hunk 2 above)
Comment (KristianSelnas) [reply src/components/map/types.ts:41]: Ok. Maybe `MapPadding` should be more like the old `MapboxGL.Padding` then? I.e. `type MapPadding = number | [number, number] | [number, number, number, number]`? I assume that the padding must still consist of a 2 or 4 array?

Comment (jorelosorio) [reply src/components/map/types.ts]: I made as you suggested ;)
